File: nim/leap/leap.nim
proc isLeapYear*(year: int): bool =
  let by4 = year mod 4 == 0
  let by100 = year mod 100 == 0
  let by400 = year mod 400 == 0

  if by4 and by100:
    by400
  else:
    by4

File: ocaml/leap/leap.ml
let is_divisible_by divisor n =
	n mod divisor == 0

let leap_year year =
	if (is_divisible_by 4 year) then
		if (is_divisible_by 100 year) then
			is_divisible_by 400 year
		else
			true
		else
			false